mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -38,6 +40,53 @@
 sqlalchemy.orm.configure_mappers()
 
 
+# similar to https://docs.sqlalchemy.org/en/13/orm/loading_relationships.html#subquery-eager-loading
+# but with (optional) filtering
+def disjoint_load(query, parents, attr, whitelist=None):
```

Comment:
This is super complex operation. Can you please explain what is happening here? It would help to add some examples in the function header to explain what each of these arguments are doing? 
It would be beneficial to understand what is happening in this method, inputs, outputs, and what is being processed.  For instance it looks like `query` is a SQL Alchemy query but it is unclear what `parents` means here! Also `attr` from the caller looks like a SQL table column but is not very clear from the function description.